sim-administration/certificate-authority-simulated/circle-simulated.sh

```diff
@@ -36,14 +40,26 @@ done
 ## misc keys, certificats and csrs.
 ##
 
+
 ARTEFACT_ROOT=crypto-artefacts
-if [[ -r "$ARTEFACT_ROOT" ]] ; then 
-   rm -f $ARTEFACT_ROOT
+
+
+# Reset if requested
+if [[ -n "$RESET_CRYPTO_ARTEFACTS" ]] ; then 
+    if [[ -f "$ARTEFACT_ROOT" ]] ; then 
+	rm -f $ARTEFACT_ROOT
+    fi
 fi
+    
+SIM_MANAGER="sim-mgr"
+SMDPPLUS="sm-dp-plus"
+
+ACTORS="$SIM_MANAGER $SMDPPLUS"
 
-ACTORS="sim-mgr sm-dp-plus"
 for  x in $ACTORS ; do 
-  mkdir -p "$ARTEFACT_ROOT/$x"
+    if [[ ! -d "$ARTEFACT_ROOT/$x" ]] ; then 
+        mkdir -p "$ARTEFACT_ROOT/$x"
```

Comment:
Minor. The `if [[` test is actually not needed as the '-p' option to mkdir turns the command into a noop if the directory already exists.